accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -315,12 +315,20 @@ or if you imported the extension with a prefix to avoid name collision:
 prefix.MyList<String>(object).quickSort();
 ```
 
-The syntax looks like a constructor invocation, but it does not create a new object.
+The syntax looks like a constructor invocation, but it does not create a new object. Instead it evaluates `object` to a value, then calls the `quickSort` extension method of `MyList` on that object.
 
 If `object.quickSort()` would invoke an extension method of `MyList`, then `MyList(object).quickSort()` will invoke the exact same method in the same way.
 
 The syntax is not *convenient*&mdash;you have to put the "constructor" invocation up front, which removes the one advantage that extension methods have over normal static methods. It is not intended as the common use-case, but as an escape hatch out of unresolvable conflicts.
 
+If the expression `MyList(object)` is evaluated for its value, it evaluates to the value of  `object`. It only works as a scope override for member lookup, similarly to how a class name works differently for static member lookup and evaluation to a `Type` object.
+
+A cascade expression like `var o = MyList(expr)..quickSort()..quickSort()` works correctly, the static members of `MyList` are invoked by the cascade, and then `o` is bound to the value of `expr`. As such, the static type of `MyList(object)` is the static type of `object`, it is just that the static type is not what is used for member lookup.
```

Comment:
I'm actually not happy with the static type here.

If, hypothetically, we want to use the current extension declarations as extension types as well, then `MyList(expr)` should have `MyList` as static type at that point, allowing only the extension members to be used. If we give it type `List` now, then someone might start doing `(MyList(expr)..extensionMember()).listMember()`, and that would *stop* working if `MyList(expr)` changes static type.

So, for now I'll recommend that the static type of `MyList(expr)` as a value expression is `void`. You can still do `MyList(expr).foo()` or `MyList(expr)..foo()..bar()` because lookups won't use the static type (just like `ClassName.foo` does static member lookup, but the static type of `ClassName` by itself is `Type`).

We should discourage `MyList(expr)` as a value expression anywhere other than in a cascade because it's just equivalent to `expr as void`, but the `void` type would prevent users from using the value of the cascade for now, until we perhaps get extension types.

Reply:
Not sure how I feel about this.  It's definitely not what I would do ignoring the extension type possibility, and does take away a useful affordance.  It will also be quite surprising to users, I think, if they encounter it.  On the other hand, it's a pretty corner case thing.

Do you really expect to want to piggy back extension types on this?  It feels unlikely to work out well in general to me, but maybe.